code/modules/mob/living/simple_animal/hostile/pulse_demon/pulsedemon_powers.dm

```diff
@@ -646,6 +646,26 @@
 		PD.current_robot = null
 		PD.current_bot = null
 		PD.current_weapon = null
+		PD.change_sight(removing = SEE_TURFS | SEE_MOBS | SEE_OBJS)
 	else
 		owner.forceMove(get_turf(owner))
 	Remove(owner)
+
+/datum/action/pd_change_camera
+	name = "Change camera view"
+	desc = "Move to another camera in the area."
+	icon_icon = 'icons/mob/screen_ai.dmi'
+	button_icon_state = "camera"
+
+/datum/action/pd_change_camera/Trigger()
+	if(ispulsedemon(owner))
+		var/mob/living/simple_animal/hostile/pulse_demon/PD = owner
+		var/list/camstouse = list()
+		var/area/ourarea = get_area(PD)
+		for (var/obj/machinery/camera/C in cameranet.cameras)
+			if(get_area(C) == ourarea)
+				camstouse["[C.c_tag]"] += C
```

Comment:
If possible, search for cameras in your area only instead of checking get_area on every camera in the camera net.

Reply:
would kinda require making another list in the /area type to iterate over because i think there's a lot more objects per area on average than cams in the world, so bigger loop